lib/internal/modules/esm/fetch_module.js

```diff
@@ -89,6 +89,18 @@ function createUnzip() {
   return createUnzip();
 }
 
+/**
+ * Redirection status code as per section 6.4 of RFC 7231:
+ * https://datatracker.ietf.org/doc/html/rfc7231#section-6.4
+ * and RFC 7238:
+ * https://datatracker.ietf.org/doc/html/rfc7238
+ * @param {number} statusCode
+ * @returns {boolean}
+ */
+function isRedirect(statusCode) {
+  return [300, 301, 302, 303, 307, 308].includes(statusCode);
```

Comment:
Unfortunately, within the Node codebase we have to use [primordials](https://github.com/nodejs/node/blob/main/doc/contributing/primordials.md) for prototype methods like `Array.prototype.includes`. Or we can use more basic syntax:

```suggestion
  switch (statusCode) {
    case 300: // Multiple Choices
      return true;
    case 301: // Moved Permanently
      return true;
    case 302: // Found
      return true;
    case 303: // See Other
      return true;
    case 307: // Temporary Redirect
      return true;
    case 308: // Permanent Redirect
      return true;
    default:
      return false;
  }
```

Reply:
The verbosity was so you could include comments explaining each status, since the previous author and reviewers of this code weren’t aware of all of them. But I don’t particularly care either way.

Reply:
@ljharb How about [this](https://github.com/nodejs/node/compare/main...hemanth:node:fetch-redirects?expand=1#diff-90074cf1270f0cf5c7c280398d290c54eaac7c9f99f7d130fa4beb5548e015a9R111)?

Reply:
If we're using an array or a Set, it should be defined outside of the function body so it doesn't have to be instantiated at each call. Note that using an array or a Set doesn't prevent us from adding comments, e.g.:

```js
const redirectHTTPStatusCodes = [
  300, // Multiple Choices
  301, // Moved Permanently
  // ...etc.
];
// Or with a Set:
const redirectHTTPStatusCodes = new SafeSet()
  .add(300) // Multiple Choices
  .add(301) // Moved Permanently
  // ...etc.
  ;
```

Reply:
I would have a slight preference for an array/Set declared outside. If you want the comments like Geoffrey suggested, that's still possible:

```
const redirectHTTPStatusCodes = new SafeSet([
  300, // Multiple Choices
  301, // Moved Permanently
  // …
]);
```

Or, merely a between:

```
if (statusCode >= 300 && statusCode < 400)
```

I'm a little surprised the original implementation didn't account for other 3XX statuses. I can't think why to exclude them, but it makes me I wonder if that was intentional 🤔 Unfortunately, the original author is on sabbatical.

Reply:
Why are we skipping 304?

Reply:
> Why are we skipping 304?

See https://github.com/nodejs/node/pull/43689#discussion_r915087791.

> I'd like to point out that `3xx` codes are not only redirects: for example, there is `304 Not Modified`.